newerhoods/server.R

```diff
@@ -176,15 +177,25 @@ function(input, output) {
     newerhoods <- SpatialPolygonsDataFrame(newerhoods,newerhoods_df)
 
     ## Add the ability to download the results as GeoJSON
-    output$downloadData <- downloadHandler(
+    output$downloadGEOJson<- downloadHandler(
         filename = function() {
-            paste("clusters",".geojson",sep="") 
-        }, 
+            paste("clusters",".geojson",sep="")
+        },
         content = function(file){
             write(as.geojson(newerhoods), file )
         }
     )
 
+
+    output$downloadPNG<- downloadHandler(
+        filename = function() {
+            paste("clusters",".png",sep="")
+        },
+        content = function(file){
+            mapshot(map_reactive(), file=file)
```

Comment:
This works, but is pretty slow. I am wondering if instead of exporting the leaflet map by using a snapshot method, it might be better to use R's plotting abilities to construct the image?

Reply:
I'll merge this for now and commit changes separately to plot using `ggmap` as I did for the Research Ex animation. It'll be faster that way.